### PR TITLE
change isinstance

### PR DESCRIPTION
**Pull request by Rubtsowa**

### CODE REVIEW

File: sdc/datatypes/hpat_pandas_dataframe_functions.py [hunk 1]
```diff
@@ -1995,16 +1995,16 @@ def pct_change_overload(df, periods=1, fill_method='pad', limit=None, freq=None)
     ty_checker = TypeChecker('Method {}().'.format(name))
     ty_checker.check(df, DataFrameType)
 
-    if not isinstance(periods, (types.Integer, types.Omitted)):
+    if not isinstance(periods, (types.Integer, types.Omitted)) and periods != 1:
```
Comment (densmirn) [inline sdc/datatypes/hpat_pandas_dataframe_functions.py:1998]: Why `periods` could not be equal to other values?

File: sdc/datatypes/hpat_pandas_dataframe_functions.py [hunk 2]
```diff
@@ -1995,16 +1995,16 @@ def pct_change_overload(df, periods=1, fill_method='pad', limit=None, freq=None)
...
 
     if not isinstance(fill_method, (str, types.UnicodeType, types.StringLiteral, types.NoneType, types.Omitted)):
         ty_checker.raise_exc(fill_method, 'string', 'fill_method')
 
-    if not isinstance(limit, (types.Omitted, types.NoneType)):
+    if not isinstance(limit, (types.Omitted, types.NoneType)) and limit is not None:
         ty_checker.raise_exc(limit, 'None', 'limit')
 
-    if not isinstance(freq, (types.Omitted, types.NoneType)):
+    if not isinstance(freq, (types.Omitted, types.NoneType)) and freq is not None:
```
Comment (densmirn) [inline sdc/datatypes/hpat_pandas_dataframe_functions.py:2007]: Please add unit test on situation when `limit` and `freq` are equal to `None` on the typing stage. Pay your attention that there are no such checks in `Series.pct_change` overload.

File: sdc/datatypes/hpat_pandas_dataframe_functions.py [hunk 3]
```diff
@@ -1954,7 +1954,7 @@ def pct_change_overload(df, periods=1, fill_method='pad', limit=None, freq=None)
 
     Limitations
     -----------
-    Parameters ``limit`` and ``freq`` are unsupported.
+    Parameters ``limit`` and ``freq`` are supported.
```
Comment (densmirn) [inline sdc/datatypes/hpat_pandas_dataframe_functions.py:1957]: This statement doesn't look like a limitation.

File: sdc/datatypes/hpat_pandas_dataframe_functions.py (hunk 1 above)
Comment (Rubtsowa) [reply sdc/datatypes/hpat_pandas_dataframe_functions.py:1998]: @densmirn  ``periods`` could be equal to other values. This condition need that default value supported.
